addons/website/static/src/components/navbar/navbar.js

```diff
@@ -1,35 +1,40 @@
 /** @odoo-module **/
 
 import { NavBar } from '@web/webclient/navbar/navbar';
-import { useService, useBus } from '@web/core/utils/hooks';
+import { useService } from '@web/core/utils/hooks';
 import { registry } from "@web/core/registry";
 import { patch } from 'web.utils';
 
 const websiteSystrayRegistry = registry.category('website_systray');
 
+const { useState, useEffect } = owl;
+
 patch(NavBar.prototype, 'website_navbar', {
     setup() {
         this._super();
         this.websiteService = useService('website');
         this.websiteCustomMenus = useService('website_custom_menus');
 
-        // The navbar is rerendered with an event, as it can not naturally be
-        // with props/state (the WebsitePreview client action and the navbar
-        // are not related).
-        useBus(websiteSystrayRegistry, 'EDIT-WEBSITE', () => this.render(true));
-
         if (this.env.debug && !websiteSystrayRegistry.contains('web.debug_mode_menu')) {
             websiteSystrayRegistry.add('web.debug_mode_menu', registry.category('systray').get('web.debug_mode_menu'), {sequence: 100});
         }
 
-        useBus(websiteSystrayRegistry, 'CONTENT-UPDATED', () => this.render(true));
+        this.currentWebsite = useState(this.websiteService.currentWebsite);
+
+        useEffect(isLoaded => {
+            // When the website is loaded, all the website systray elements
+            // have been rendered and the navbar must be adapted accordingly.
+            if (isLoaded) {
+                this.adapt();
+            }
+        }, () => [this.isWebsiteLoaded()]);
```

Comment:
Just want to make sure I understand cause I'm intrigues by this one:

The reactive value `this.websiteService.website` is used in the component, which means that when it changes, the dependency array will be computed. If its value is different, then the effect is executed. Is that correct?

If so maybe a comment is worth explaining why the length of the metadata keys is used. (I assume it's cause they're either all here and a website is present, or none of them are and no website is loaded / it's not an editable page)

I'm mostly curious about it all though!

Reply:
Yes:
- the component uses a state on `websiteService.currentWebsite`: `this.currentWebsite`
- the value of the key `metadata` of this state is read in the component's isWebsiteLoaded method
=> the component will be rerendered when that value changes
=> when the component is rerendered, the dependency of the effect will be compared to the previous render's one 
=> it will apply the effect if it changed
It is definitely worth a comment, thanks for pointing it out !